app/components/Sidebar.tsx

```diff
@@ -88,20 +89,24 @@ export const NavLinkItem = (props: {
   children: React.ReactNode
   end?: boolean
   disabled?: boolean
-}) => (
-  <li>
-    <NavLink
-      to={props.to}
-      className={({ isActive }) =>
-        cn(linkStyles, {
-          'text-accent !bg-accent-secondary hover:!bg-accent-secondary-hover svg:!text-accent-tertiary':
-            isActive,
-          'pointer-events-none text-disabled': props.disabled,
-        })
-      }
-      end={props.end}
-    >
-      {props.children}
-    </NavLink>
-  </li>
-)
+}) => {
+  const location = useLocation()
+  const isLinkToRootResource = isRootResourceLink(props.to, location.pathname)
```

Comment:
I see why you picked the name `isRootResourceLink`, but to me, because it’s inside the nav link component, it’s kind of backwards. This name says: from the point of view of the create route, is this link relevant to me due to being my root resource? But we’re inside the link here, not the create form, so to me the more intuitive way to put it would be: from the point of view of this link, are we currently on the corresponding create form route? Hence `onCreatePage` in my other suggestion. Could be `onCreateFormRoute` or whatever.